src/Microsoft.ManagedZLib/src/TreeDesc.cs

```diff
@@ -0,0 +1,310 @@
+        // The last code must be all ones.
+        Debug.Assert(code + codeCount[DeflateTrees.MaxBits] - 1 == (1 << DeflateTrees.MaxBits) - 1,
+                "inconsistent bit counts");
+
+        for (n = 0; n <= _maxCode; n++)
+        {
+            int len = _dynamicTree[n].Len;
+            if (len == 0) continue;
+            /* Now reverse the bits */
+            _dynamicTree[n].Freq = (ushort)BitReverse(nextCode[len]++, len);
+        }
+    }
+
+    // Index within the heap array of least frequent node in the Huffman tree
+    public const int Smallest = 1;
+
+    // Remove the smallest element from the heap and recreate the heap with
+    // one less element.Updates heap and heap_len.
+    public void PriorityQueueRemove(out int removedTop, int[] Heap, ref int heapLen, byte[] depth)
+    {
+        removedTop = Heap[Smallest];
+        Heap[Smallest] = Heap[heapLen--];
+        PriorityQueueDownHeap(Smallest, Heap, ref heapLen, depth);
+    }
+
+    // Compares to subtrees, using the tree depth as tie breaker when
+    // the subtrees have equal frequency. This minimizes the worst case length.
+    public static bool Smaller(CtData[] tree, int n, int m, byte[] depth) =>
+        (tree[n].Freq < tree[m].Freq ||
+        (tree[n].Freq == tree[m].Freq && depth[n] <= depth[m]));
+
+    // Restore the heap property by moving down the tree starting at node (parameter)"node",
+    // exchanging a node with the smallest of its two sons if necessary,
+    //  stopping when the heap property is re-established
+    // (each father smaller than its two sons).
+    // tree : The tree to restore 
+    // node: Node to move down
+    public void PriorityQueueDownHeap(int node, int[] Heap, ref int heapLen,byte[] depth)
+    {
+        int v = Heap[node];
+        int j = node << 1;  /* left son of node */
+        while (j <= heapLen)
+        {
+            /* Set j to the smallest of the two sons: */
+            if (j < heapLen &&
+                Smaller(_dynamicTree, Heap[j + 1], Heap[j], depth))
+            {
+                j++;
+            }
+            /* Exit if v is smaller than both sons */
+            if (Smaller(_dynamicTree, v, Heap[j], depth)) break;
+
+            /* Exchange v with the smallest son */
+            Heap[node] = Heap[j]; node = j;
+
+            /* And continue down the tree, setting j to the left son of k */
+            j <<= 1;
+        }
+        Heap[node] = v;
+    }
+
+
+
+    public void GenBitLen(Span<int> Heap,
+                          Span<ushort> _codeCount,
+                          ref ulong _optLength,
+                          ref ulong _staticLen,
+                          ref int _heapMax)
+    {
+        CtData[] tree = _dynamicTree!;
+        int max_code = _maxCode;
+        CtData[] STree = _StaticTreeDesc.staticTree!;
+        int[] Extra = _StaticTreeDesc.extraBits!;
+        int Base = _StaticTreeDesc.extraBase;
```

Comment:
These are local variables, should start with lowercase:
```suggestion
        CtData[] sTree = _StaticTreeDesc.staticTree!;
        int[] extra = _StaticTreeDesc.extraBits!;
        int base = _StaticTreeDesc.extraBase;
```